src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs

```diff
@@ -95,6 +94,30 @@ public virtual async Task<OrderHistoryViewModel> GetOrderHistoryViewModelAsync(G
             {
                 shipmentsTrackingInfos = await GetShipmentsTrackingInfoViewModels(orderQueryResult, param).ConfigureAwait(false);
             }
+            var orderSettings = await GetOrderSettings(param.Scope).ConfigureAwait(false);
+            
+            var fulfillmentOrdersList = new List<FulfillmentOrderQueryResult>();
+            if (param.OrderTense == OrderTense.CurrentOrders)
+            {
+                var fulfillmentLocationsToOrder = orderQueryResult.Results
+                    .Select(item => new
+                    {
+                        fulfillmentLocationIds = item.ShipmentItems.Select(el => el.FulfillmentLocationId).ToList(),
+                        orderId = item.Id
+                    }).ToList();
+
+                var fulfillmentLocationToOrder = fulfillmentLocationsToOrder
+                    .SelectMany(el => el.fulfillmentLocationIds.Select(item => new {orderId = el.orderId, fulfillmentLocationId = item}))
+                    .Distinct().ToList();
+                fulfillmentLocationToOrder
+                    .GroupBy(item => item.fulfillmentLocationId)
+                    .ToList()
+                    .ForEach(item => 
+                        fulfillmentOrdersList.Add(OrderRepository.FindFulfillmentOrders(param.Scope, item.Select(el => Guid.Parse(el.orderId)).ToList(), item.Key).Result));
+                
```

Comment:
Please use await with ConfigureAwait(false) to avod deadlocks
You can rewrite your code like this and use await Task.WhenAll:

```
fulfillmentOrdersList = await Task.WhenAll(fulfillmentLocationToOrder
.GroupBy(item => item.fulfillmentLocationId)
.Select(async item =>
await OrderRepository.FindFulfillmentOrders(param.Scope, item.Select(el => Guid.Parse(el.orderId)).ToList(), item.Key).ConfigureAwait(false))
).ConfigureAwait(false);
```